eth/db/storage.py

```diff
@@ -0,0 +1,477 @@
+from abc import ABC, abstractmethod
+import logging
+from typing import (  # noqa: F401
+    cast,
+    Dict,
+    Iterable,
+    Set,
+    Tuple,
+)
+from uuid import UUID, uuid4
+
+from eth_hash.auto import keccak
+from eth_typing import (
+    Address,
+    Hash32
+)
+from eth_utils import (
+    ValidationError,
+    to_tuple,
+    int_to_big_endian,
+)
+import rlp
+from trie import (
+    HexaryTrie,
+    exceptions as trie_exceptions,
+)
+
+from eth._utils.padding import (
+    pad32,
+)
+from eth.constants import (
+    BLANK_ROOT_HASH,
+)
+from eth.db.account import (  # noqa: F401
+    BaseAccountDB,
+)
+from eth.db.backends.base import (
+    BaseAtomicDB,
+    BaseDB,
+)
+from eth.db.batch import (
+    BatchDB,
+)
+from eth.db.cache import (
+    CacheDB,
+)
+from eth.db.journal import (
+    JournalDB,
+)
+from eth.tools.logging import (
+    ExtendedDebugLogger
+)
+
+
+class BaseStorageDB(ABC):
```

Comment:
I guess it's `py-evm`s convention to make interfaces like this, but I wonder why? It doesn't seem likely another class will  implement this interface any time soon, and the primary effect seems to be separating these useful docstrings from the code they document.

Reply:
Some general rationals are:
- easier backend swapping during testing (don't think we use it here, but I could maybe imagine it
- preparing for things that change from one fork to the next
- easier fiddling with swapping in 
- a stronger guarantee about which methods are accessed publicly, since mypy will complain if you access anything only defined on the implementing class.

Not necessarily a good reason, but it's a reason. :)

Reply:
Yeah, that makes sense, thanks for the answer!